search-index/index_data.py

```diff
@@ -0,0 +1,248 @@
+"""Index data into a running Opensearch index."""
+
+import os
+from pathlib import Path
+from typing import Generator
+
+import pandas as pd
+import numpy as np
+import click
+
+from navigator.core.log import get_logger
+from app.db import PostgresConnector
+from app.index import OpenSearchIndex
+
+logger = get_logger(__name__)
+
+postgres_connector = PostgresConnector(os.environ["DATABASE_URL"])
+
+
+def get_data_from_navigator_tables() -> pd.DataFrame:
+    """Get data from Navigator tables. Includes documents, actions, countries, languages, action sources and action types.
+
+    Returns:
+        pd.DataFrame: one row per document.
+    """
+    query = """
+        SELECT document_id, source_url, s3_url, document.language_id as document_language_id, document.name AS document_name, action.*, language.language_id, language.language_code, language.name as language_name, \
+        geography.*, source.source_id, source.name as action_source_name, action_type.action_type_id, action_type.type_name as action_type_name
+        FROM document
+        INNER JOIN action ON (document.action_id = action.action_id)
+        LEFT JOIN language on (document.language_id = language.language_id)
+        LEFT JOIN geography on (action.geography_id = geography.geography_id)
+        LEFT JOIN source on (action.action_source_id = source.source_id)
+        LEFT JOIN action_type on (action.action_type_id = action_type.action_type_id);
+        """
```

Comment:
Please can someone double-check this SQL query as it's one of the first I've ever written!

Reply:
If it works for now, then great. This will change anyway, as we'll put data from external sources in a "stage 0" DB first, where the indexer will be reading from, and then data for public consumption will go to the next stage.